src/text-editor-component.js

```diff
@@ -2361,13 +2368,29 @@ class TextEditorComponent {
     })
     const didDestroyDisposable = decoration.onDidDestroy(() => {
       this.blockDecorationsToMeasure.delete(decoration)
+      this.heightsByBlockDecoration.delete(decoration)
+      this.blockDecorationsByElement.delete(element)
+      this.blockDecorationResizeObserver.unobserve(element)
       this.lineTopIndex.removeBlock(decoration)
       didUpdateDisposable.dispose()
       didDestroyDisposable.dispose()
       this.scheduleUpdate()
     })
   }
 
+  didResizeBlockDecorations (entries) {
+    if (!this.visible) return
+
+    for (let i = 0; i < entries.length; i++) {
+      const {target, contentRect} = entries[i]
+      const decoration = this.blockDecorationsByElement.get(target)
+      const previousHeight = this.heightsByBlockDecoration.get(decoration)
+      if (this.element.contains(target) && contentRect.height !== previousHeight) {
+        this.invalidateBlockDecorationDimensions(decoration)
```

Comment:
Yeah, I considered doing that but the way we measure block decorations is slightly more complex than retrieving their `offsetHeight` (e.g. [we use "sentinel elements"](https://github.com/atom/atom/blob/10a7cb5badc56ac6b5d02476fe330d6e53f1cedd/src/text-editor-component.js#L285-L333)), so it felt more consistent to use the same method across all call sites.